src/ngx_http_lua_control.c

```diff
@@ -304,6 +304,11 @@ ngx_http_lua_ngx_exit(lua_State *L)
         return luaL_error(L, "expecting one argument");
     }
 
+    rc = (ngx_int_t) luaL_checkinteger(L, 1);
```

Comment:
Yes, thanks for the reminder.

Maybe such args validation should be handled in the Lua land (in `resty.core.exit`)? My initial hunch is that it should be in `ngx_http_lua_ffi_exit` as well so I'll update this PR, but I am unsure.